src/internal_events/aws_ecs_metrics.rs

```diff
@@ -0,0 +1,98 @@
+use super::InternalEvent;
+use metrics::{counter, histogram};
+use std::borrow::Cow;
+use std::time::Instant;
+
+#[derive(Debug)]
+pub struct AwsEcsMetricsReceived {
+    pub byte_size: usize,
+    pub count: usize,
+}
+
+impl InternalEvent for AwsEcsMetricsReceived {
+    fn emit_logs(&self) {
+        debug!(message = "Scraped events.", ?self.count);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("events_processed_total", self.count as u64);
+        counter!("processed_bytes_total", self.byte_size as u64);
+    }
+}
+
+#[derive(Debug)]
+pub struct AwsEcsMetricsRequestCompleted {
+    pub start: Instant,
+    pub end: Instant,
+}
+
+impl InternalEvent for AwsEcsMetricsRequestCompleted {
+    fn emit_logs(&self) {
+        debug!(message = "Request completed.");
+    }
+
+    fn emit_metrics(&self) {
+        counter!("requests_completed_total", 1);
+        histogram!("request_duration_nanoseconds", self.end - self.start,
```

Comment:
I believe this should be `request_duration_nanoseconds` to match our naming conventions (based on https://prometheus.io/docs/practices/naming/).